app/Helpers/database/player-game.php

```diff
@@ -747,18 +747,28 @@ function getGameTopAchievers(int $gameID): array
         $numAchievementsInSet = $data['NumAchievementsInSet'];
     }
 
-    // TODO config('feature.aggregate_queries') slow query (17)
-    $query = "SELECT aw.User, COUNT(*) AS NumAchievements, SUM(ach.points) AS TotalScore, MAX(aw.Date) AS LastAward
-                FROM Awarded AS aw
-                LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
-                LEFT JOIN GameData AS gd ON gd.ID = ach.GameID
-                LEFT JOIN UserAccounts AS ua ON ua.User = aw.User
-                WHERE NOT ua.Untracked
-                  AND ach.Flags = " . AchievementFlag::OfficialCore . "
-                  AND gd.ID = $gameID
-                  AND aw.HardcoreMode = " . UnlockMode::Hardcore . "
-                GROUP BY aw.User
-                ORDER BY TotalScore DESC, NumAchievements DESC, LastAward";
+    if (config('feature.aggregate_queries')) {
+        $query = "SELECT ua.User, pg.achievements_unlocked_hardcore AS NumAchievements,
+                         pg.points_hardcore AS TotalScore, pg.last_unlock_hardcore_at AS LastAward
+                    FROM player_games pg
+                    LEFT JOIN UserAccounts ua ON ua.ID = pg.user_id
+                    WHERE NOT ua.Untracked
```

Comment:
nit:
```suggestion
                    WHERE ua.Untracked = 0
```

When using `NOT` in front of Untracked, it makes querying that data non-sargable, blocking any indexes on the `Untracked` column from being used. The `NOT` operator acts as a transformation on the column, preventing an index from being used.

Reply:
That cut another 90ms off the query time. Still about 160ms for the query.

Reply:
Other optimization ideas:

* Add `LIMIT 10`.
* Use an `INNER JOIN` instead of a `LEFT JOIN` on `UserAccounts`.

Unfortunately, I'm not able to test either of these at the moment with the state of my local database.

Reply:
We can't have a limit as it has to return all rows where users have mastered. In this case, 562:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/46ba644e-5982-4486-9f31-1d01ecf72ce6)

`INNER JOIN` did cut off another 20ms. That's a change we can probably apply to a lot of the slow queries without needing to use the aggregate tables.